feat(link): migrate link component

alexandereneroth opened the pull request:
Migrate link component

Fixes: [AB#2399](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2399)

mJarsater reviewed (approved):
Looks good to me

mJarsater commented on components/src/components/link/link.stories.ts:
```diff
@@ -0,0 +1,55 @@
+NoUnderline.args = {
+  ...Default.args,
+  noUnderline: true,
+};
```
Should this be a new story? Or would we want to streamline it and let the users view this via the controls.

alexandereneroth replied:
I feel like there's value to displaying the different variations to the user up front.
Some examples:
- https://digitaldesign.scania.com/components/datepicker/code 
- https://chakra-ui.com/docs/components/link/usage
- https://mui.com/joy-ui/react-link/

mJarsater replied:
Absolutely, I think this also is a question of how user actually use the site. Are they interacting with the controls or are they checking the stories.

theJohnnyMe reviewed (changes requested):
Please leave only the "Default" story in this one. It is a very simple component. Thanks for fixing the disabled state too.

alexandereneroth commented:
> Please leave only the "Default" story in this one. It is a very simple component. Thanks for fixing the disabled state too.

Removed the other stories. 
Also added a demo class with a font size so the text is not tiny.

mJarsater commented on components/src/components/link/link.stories.ts:
```diff
@@ -0,0 +1,48 @@
+import clsx from 'clsx';
+import { formatHtmlPreview } from '../../utils/utils';
+
+const defaultArgs = {
```
Is this object needed? The default values are already present in the default export.
